cost/azure/databricks/rightsize_compute/azure_databricks_rightsize_compute.pt

```diff
@@ -0,0 +1,1478 @@
+#   end
+#   result do
+#     encoding "json"
+#     collect jmes_path(response, "warehouses[*]") do
+#       field "databricks_workspace", iter_item
+#       field "warehouse", col_item
+#     end
+#   end
+# end
+
+# datasource "ds_databricks_instancepools" do
+#   iterate $ds_databricks_workspaces_filtered
+#   request do
+#     auth $auth_databricks
+#     host val(iter_item, "workspaceUrl")
+#     path "/api/2.0/instance-pools/list"
+#   end
+# end
+
+# datasource "ds_databricks_clusters_events" do
+#   iterate $ds_databricks_workspaces_filtered
+#   request do
+#     auth $auth_databricks
+#     pagination $pagination_databricks_events
+#     host jq(iter_item, ".databricks_workspace.workspaceUrl")
+#     path "/api/2.0/clusters/events"
+#     verb "POST"
+#     body_field "cluster_id", jq(iter_item, ".cluster.cluster_id")
+#     body_field "event_types", ["DRIVER_HEALTHY"] # DRIVER_HEALTHY when a "driver" aka instance is started and checks-in as ready
+#     body_field "limit", 2
+#   end
+#   result do
+#     encoding "json"
+#     collect jmes_path(response, "events[*]") do
+#       field "databricks_workspace", val(iter_item, "databricks_workspace")
+#       field "cluster", val(iter_item, "cluster")
+#       field "event", col_item
+#     end
+#   end
+# end
+
+datasource "ds_filtered_workspace_virtualmachines" do
+  run_script $js_filtered_workspace_virtualmachines, $ds_databricks_workspaces_filtered, $ds_azure_databricks_managed_vm_metrics, $param_stats_threshold, $param_stats_underutil_threshold_cpu_value, $ds_databricks_workspaces_vm_costs, $ds_lookback_time_formats, $ds_databricks_clusters, $ds_azure_sku_memory
+end
+
+script "js_filtered_workspace_virtualmachines", type: "javascript" do
+  parameters "ds_databricks_workspaces", "ds_azure_databricks_managed_vm_metrics", "param_stats_threshold", "param_stats_underutil_threshold_cpu_value", "ds_databricks_workspaces_vm_costs", "ds_lookback_time_formats", "ds_databricks_clusters", "ds_azure_sku_memory"
+  result "result"
+  code <<-EOS
+  var lastTick = 0; // Holds the timestamp last tick was called
+  var tickCount = 0; // Holds the count tick was called
+  // var tickInterval = 10; // Only log one out of every X ticks
+  function debugLogSample(msg) {
+    // Default sinceLastTick to 0 -- we can calculate this later if we have a lastTick set
+    var sinceLastTick=0
+    // Only print one out of every 10 ticks
+    if (tickCount % 10 == 0) {
+      // Log the lastTick
+      if (lastTick > 0) {
+        sinceLastTick = Date.now() - lastTick
+      }
+      // Print debug
+      console.log("sinceLastTickMs="+sinceLastTick+" "+msg)
+      // Update lastTick
+      lastTick = Date.now()
+    }
+    // Increment tick every time
+    // Do this last so the first log printed happens on the first tick
+    tickCount = tickCount+1;
+  }
+
+  // function computeConfidenceInterval(observations, confidenceLevel, zScore) {
```

Comment:
Defunct code should be removed unless it's here with the intent of being used at some point in the future